lib/buffer.js

```diff
@@ -379,6 +378,16 @@ Buffer.prototype.toString = function(encoding, start, end) {
 };
 
 
+Buffer.prototype.toString = function() {
+  const length = this.length | 0;
+  if (length === 0)
+    return '';
+  if (arguments.length === 0)
+    return this.utf8Slice(0, length);
+  return slowToString.apply(this, arguments);
```

Comment:
@trevnorris What do you mean cleaning up that particular section of code? That code is switching the arguments length and only passing that many arguments.

FWIW I already benchmarked various alternative function calling methods for this patch on top of the next branch:

Replacing `apply()` with `.call(this, arguments[0], arguments[1], ...)` slows down the cases when there are arguments passed, and there is a slight performance hit in the zero argument case (with apply I saw ~510% increase, but call showed ~470%).

Replacing `apply()` with a direct function call, passing in the context as an extra argument performs about the same as using `.call()`.

Replacing `apply()` with a switch on `arguments.length` and using either `.call()` or passing the context in the < 3 cases (using `.apply()` as default), the zero argument case is a bit lower IIRC (~470% increase), but now the non-zero argument cases are no longer affected.

So just using `.apply()` instead of several-line switch is shorter and even a tad faster on the zero argument case.